scripts/gen-auth-api-spec.ts

```diff
@@ -130,6 +128,70 @@ async function toOpenapi3(discovery: any): Promise<any> {
   return openapi3;
 }
 
+const pathParamsForFlatPathParam: Map<string, string> = new Map([
+  ["{projectsId}", "{targetProjectId}"],
+  ["{tenantsId}", "{tenantId}"],
+]);
+
+const paramPattern = /{([^}]+)}/g;
+
+function replaceWithFlatPath(discovery: { [key: string]: any }): void {
+  if (discovery.methods) {
+    Object.entries(discovery.methods).forEach(([_, method]) => {
+      if (typeof method === "object") {
+        // Replace flat path param names with path param names
+        // e.g. for endpoint identitytoolkit.projects.defaultSupportedIdpConfigs.get:
+        // path = v2/{name}
+        // flatPath = v2/projects/{projectsId}/defaultSupportedIdpConfigs/{defaultSupportedIdpConfigsId}
+        //
+        // transform v2/projects/{projectsId}/defaultSupportedIdpConfigs/{defaultSupportedIdpConfigsId}
+        //       --> v2/projects/{targetProjectId}/defaultSupportedIdpConfigs/{defaultSupportedIdpConfigsId}
+        let flatPath = (method as any).flatPath;
+        pathParamsForFlatPathParam.forEach((pathParam, flatPathParam) => {
+          flatPath = flatPath.replace(flatPathParam, pathParam);
+        });
+
+        const cleanedParams: { [key: string]: any } = {};
+
+        // Get all param names in path
+        // e.g. ["projectsId", "defaultSupportedIdpConfigsId"]
+        const pathParamMatches = [...flatPath.matchAll(paramPattern)];
+        const paramsInPath = pathParamMatches.map((match) => match[1]);
+
+        // Remove method path parameters that don't appear in the path
+        // e.g. remove parameter "name" that appears in original path
+        const params = (method as any).parameters;
+        Object.entries(params).forEach(([name, paramObj]) => {
+          if ((paramObj as any).location !== "path" || paramsInPath.includes(name)) {
+            cleanedParams[name] = paramObj;
+          }
+        });
+
+        // Add params that are in path but are not in the parameters object
+        // e.g. add "targetProjectId" and "defaultSupportedIdpConfigsId"
+        paramsInPath.forEach((param) => {
+          if (!Object.keys(cleanedParams).some((name) => name === param)) {
+            cleanedParams[param] = {
+              location: "path",
+              required: true,
+              type: "string",
+            };
+          }
+        });
+
+        (method as any).parameters = cleanedParams;
```

Comment:
It would probably be worth creating some sort of interface to cast `method` to instead

Reply:
Added in `Method` interface